lib/index.d.ts

```diff
@@ -122,7 +122,7 @@ export interface Output {
     /**
      * An object containing any HTTP headers where each key is a header name and value is the header content
      */
-    headers: object;
+    headers: Partial<Record<string, string>>;
```

Comment:
Header entries can contain other valid values than strings. Maybe use the [`http.OutgoingHttpHeaders`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0a800f58b8df8b26f8f9e7ad00c3fbd6ee31a602/types/node/v12/http.d.ts#L73-L75) type from node?

Reply:
I've updated this, but note the `tsconfig.json` also added to make the internal typing actually account for `undefined` (without it the `undefined` gets silently discarded). I put that as using the lib `@tsconfig/recommended` to reduce the direct boilerplate (basically any ongoing Node 8 Typescript project is going to end up with the settings they use anyway), but it can be done without the extra dependency by just copying the settings across.

Reply:
The config use for the type check, is [defined in lab](https://github.com/hapijs/lab/blob/14d4d6b0dbc40528e7e83e57418bd2c1b96e022a/lib/modules/types.js#L16-L23), which already includes `strict: true`.

The tsconfig you added is only useful for the editor. While it might make sense to include something like that, it is outside the scope of this PR, and I would much prefer if not included here.

Reply:
I've removed that.